apps/accounts/views.py

```diff
@@ -132,3 +132,16 @@ class ProfileView(JWTProtectedView, generics.RetrieveAPIView):
 
     def retrieve(self, request, *args, **kw):
         return Response(self.get_serializer(request.user).data)
+
+
+class AccountSourceView(generics.RetrieveAPIView):
+    serializer_class = AccountSourceSerializer
+
+    @waffle_switch('fxa-auth')
+    def retrieve(self, request, *args, **kwargs):
+        email = request.GET.get('email')
+        try:
+            user = UserProfile.objects.get(email=email)
+        except UserProfile.DoesNotExist:
+            user = UserProfile(fxa_id='sure')
```

Comment:
Are you `sure`? 

Jokes aside, shouldn't this be a unique ID? What is it used for exactly?

The fact that this sets `source: "fxa"` for non-existing users seems a little confusing to me. Couldn't it just set `source: null` and then the frontend could make the decision to make the source "fxa"?